js/hview.js

```diff
@@ -828,6 +853,25 @@ $(function () {
         $("#vlist_content").showError(msg ? msg : "Error loading view hierarchy")
     }
 
+    function updateNodeRelationships(node) {
+        const el = document.getElementById(node.treeDisplayName)
+        node.disablePreview = el.node.disablePreview
+        el.node = node
+        el.box.node = node
+        node.el = el
+        node.box = el.box
+
+        if (node.parent) {
```

Comment:
I am not sure you need to change the parent node ever. In what case it turns out to be different?

Reply:
There are parts of the code (like hovering over the box preview), where the child's parent is recursively called going up the stack to expand the tree node view labels.

If the parent node isn't updated, then when the box node might be pointing to a node parent that never had an "el" or "box" attached to it, causing an NPE.